sound/soc/sof/sof-audio.c

```diff
@@ -310,7 +310,7 @@ sof_unprepare_widgets_in_path(struct snd_sof_dev *sdev, struct snd_soc_dapm_widg
 	struct snd_soc_dapm_path *p;
 
 	/* return if the widget is in use or if it is already unprepared */
-	if (!swidget->prepared || swidget->use_count > 1)
+	if (!swidget || !swidget->prepared || swidget->use_count > 1)
```

Comment:
and this one is being changed by @bardliao in https://github.com/thesofproject/linux/pull/4022

Reply:
drop this patch?

Reply:
We still need to have the !swidget check